### PR TITLE
[core] Remove unused GCS RPCs

### PR DESCRIPTION
**Pull request by vitsai**
## Why are these changes needed?

Clean up some tech debt as part of test refactoring for https://github.com/ray-project/ray/pull/40488

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 1]
```diff
@@ -390,11 +390,6 @@ void LocalResourceManager::OnResourceOrStateChanged() {
   resource_change_subscriber_(ToNodeResources());
 }
 
-void LocalResourceManager::ResetLastReportResourceUsage(
-    const NodeResources &replacement) {
-  last_report_resources_.reset(new NodeResources(replacement));
```
Comment (jjyao) [inline src/ray/raylet/scheduling/local_resource_manager.cc:395]: last_report_resources_ variable can be removed as well I think

File: src/ray/gcs/gcs_server/gcs_server.h [hunk 2]
```diff
@@ -110,6 +110,11 @@ class GcsServer {
   static constexpr char kInMemoryStorage[] = "memory";
   static constexpr char kRedisStorage[] = "redis";
 
+  GcsResourceManager &GetGcsResourceManager() {
```
Comment (jjyao) [inline src/ray/gcs/gcs_server/gcs_server.h:113]: Maybe instead of exposing this, add a `UpdateNodeResourceUsageForTesting` method?

**Review by jjyao — COMMENTED** [review]
nice.

**Review by rickyyx — APPROVED** [review]
Nice! Were this unsed for a long time? 

Do you know what's the previous usage?

**Comment by vitsai** [comment]
These seem to be legacy RPCs from before we moved to Ray Syncer

File: src/ray/gcs/gcs_client/test/gcs_client_test.cc [hunk 3]
```diff
@@ -454,6 +452,14 @@ class GcsClientTest : public ::testing::TestWithParam<bool> {
...
   std::unique_ptr<instrumented_io_context> client_io_service_;
   std::unique_ptr<gcs::GcsClient> gcs_client_;
+  std::unique_ptr<std::thread> syncer_io_service_thread_;
+  std::unique_ptr<instrumented_io_context> syncer_io_service_;
+
+  // Ray Syncer and reporter.
+  std::unique_ptr<syncer::RaySyncer> ray_syncer_;
+  std::unique_ptr<syncer::RaySyncerService> ray_syncer_service_;
+  std::unique_ptr<grpc::Server> syncer_server_;
+  std::unique_ptr<LocalResourceManager> local_resource_manager_;
```
[the remark this answers is not in the document]
Comment (vitsai) [reply src/ray/gcs/gcs_client/test/gcs_client_test.cc:462]: Yeah, this was leftover from a bad idea

**Review by jjyao — APPROVED** [review]

**Pull request merged**